openni2_camera/src/openni2_device_manager.cpp

```diff
@@ -67,6 +68,7 @@ class OpenNI2DeviceListener : public openni::OpenNI::DeviceConnectedListener,
       openni::OpenNI::DeviceDisconnectedListener(),
       openni::OpenNI::DeviceStateChangedListener()
   {
+    error_state_publisher_ = nh_.advertise<std_msgs::Int32>("error_state",100);
```

Comment:
This should be a latched topic:
```suggestion
    error_state_publisher_ = nh_.advertise<std_msgs::Int32>("error_state", 10, true);
```